### DISCUSSION TITLE
How to split a string to two other variable? (for time parsing)

### DISCUSSION THREAD

**Discussion by mobin-2008** [mesonbuild/meson#11786] [Q&A] [answered]
Hi. I don't think it's a good place for my question but seems like it's.
I have a variable called `a`, it's have its options. I need to split it if it's have `.` or `,` to two variables called `a_sec` and `a_nsec`:
1. `a` variable have `.` or `,`
2. Split first part into `a_sec` and second part into `a_nsec`:
    ```
    10   .   25
    ^^   ^   ^^
    first    second
    
    a_sec = 10
    a_nsec = 25
    ```
3. Pass those variables to `config.h` via `conf.set()`

Also See my example in C++:
```c++
#include <iostream>
#include <limits> 
#include <string>

// Parse a time, specified as a decimal number of seconds (with optional fractional component after decimal
// point or decimal comma).
int parse_time(const std::string &paramval, const char * paramname, bool nano)
{
    unsigned int isec = 0;
    unsigned int insec = 0;
    auto max_secs = std::numeric_limits<decltype(isec)>::max() / 10; // I think it's can't implemented in meson
    auto len = paramval.length();
    decltype(len) i;
    for (i = 0; i < len; i++) {
        char ch = paramval[i];
        if (ch == '.' || ch == ',') {
            i++;
            break;
        }
        if (ch < '0' || ch > '9') {
            std::cerr << "mconfig-gen: " << std::string("bad value for ") + paramname << "\n";
            exit(1);
        }
        // check for overflow
        if (isec >= max_secs) {
            std::cerr << "mconfig-gen: " << std::string("too-large value for ") + paramname << "\n";
            exit(1);
        }
        isec *= 10;
        isec += ch - '0';
    }
    decltype(insec) insec_m = 100000000; // 10^8
    for ( ; i < len; i++) {
        char ch = paramval[i];
        if (ch < '0' || ch > '9') {
            std::cerr << "mconfig-gen: " << std::string("bad value for ") + paramname << "\n";
            exit(1);
        }
        insec += (ch - '0') * insec_m;
        insec_m /= 10;
    }
    if (!nano) return isec;
    return insec;
}
```
I didn't find any function to helps me to do that. I need to add external script for do that?

Regards

**Comment by eli-schwartz** [comment]
I guess what you want here is string splitting? https://mesonbuild.com/Syntax.html#split-join

**Reply by mobin-2008** [reply]
I knew that but my problem is it: "It's can't split variable into two other variables" it's just split string to some members of array and it's not suitable for my work.

**Reply by mobin-2008** [reply]
It's splinted into ['1', '0', '.', '2', '5'] instead of ['10', '.', '25']

**Reply by eli-schwartz** [reply]
The default split delimiter if unspecified splits into characters, but your use case wants to use `'.'` as the delimiter. It's described in the examples...

**Reply by mobin-2008** [reply]
Hmm, I did that already but seems like I don't use it correctly:
```meson
## Try to split something:
a = '25.62'
a.split('.')
message(a[0])
```
result
```
...
Message: 2
...
```
but i want it's be 25 instead of 2.

**Reply by eli-schwartz** [reply, marked as answer]
```meson
a_parts = a.split('.')
message(a_parts[0])
message(a[0])
```

Objects are immutable, the split method doesn't operate on "a" and turn "a" into an array -- the split method returns a value that needs to be assigned.

**Reply by mobin-2008** [reply]
I forgot variables in meson are immutable, Sorry :(
Thanks for your help.

Regards

**Reply by eli-schwartz** [reply]
No problem, happy to help. :)
